src/ports/web/guards.rs

```diff
@@ -1,127 +1,159 @@
+use crate::{
+    core::db::OrganizationRepo, core::prelude::*, core::usecases, infrastructure::error::AppError,
+    ports::web::jwt,
+};
 use chrono::prelude::*;
 use rocket::{
     self,
     http::Status,
-    outcome::IntoOutcome,
     request::{self, FromRequest, Request},
-    Outcome,
+    Outcome, State,
 };
 use std::time::Duration;
 
 pub const COOKIE_EMAIL_KEY: &str = "ofdb-user-email";
 pub const COOKIE_CAPTCHA_KEY: &str = "ofdb-captcha";
+pub const MAX_CAPTCHA_TTL: Duration = Duration::from_secs(120);
 
-#[derive(Debug)]
-pub struct Bearer(pub String);
+type Result<T> = std::result::Result<T, AppError>;
 
-impl<'a, 'r> FromRequest<'a, 'r> for Bearer {
-    type Error = ();
-    fn from_request(request: &'a Request<'r>) -> request::Outcome<Self, ()> {
-        match request.headers().get_one("Authorization") {
-            Some(auth) => {
-                let x: Vec<_> = auth.split(' ').collect();
-                if x.len() != 2 {
-                    return Outcome::Failure((Status::Unauthorized, ()));
-                }
-                if x[0] != "Bearer" {
-                    return Outcome::Failure((Status::Unauthorized, ()));
-                }
-                Outcome::Success(Bearer(x[1].into()))
-            }
-            None => Outcome::Forward(()),
-        }
+fn get_bearer_token(auth_header_val: &str) -> Option<&str> {
+    let x: Vec<_> = auth_header_val.split(' ').collect();
+    if x.len() == 2 && x[0] == "Bearer" {
+        Some(x[1])
+    } else {
+        None
     }
 }
 
 #[derive(Debug)]
-pub struct Account(String);
+pub struct Auth {
+    bearer_tokens: Vec<String>,
+    account_email: Option<String>,
+    has_captcha: bool,
+}
 
-impl Account {
-    pub fn email(&self) -> &str {
-        &self.0
+impl Auth {
+    // pub fn unauthorized() -> AppError {
```

Comment:
can we remove this?

Reply:
yeah, the unauthorized() method is not needed, forgot to remove it.

The simple `Account` implementation based on `Auth` was done because the frontend code depends on it a lot. As it's going to be removed anyway soon, I didn't bother to change everything there. When the frontend is replaced by seed app, `Account` can be removed.